### PR TITLE
Update `unique_xxx` (nep52)

### PR DESCRIPTION
**Pull request by EarlMilktea**
### NOTE

- This PR is partly related to #8531 (only `unique_xxx` )
- Need to use `with_requires("numpy>=2.1"`) because some breaking changes introduced in v2.0.0 are reverted

### CODE REVIEW

File: tests/cupy_tests/manipulation_tests/test_add_remove.py [hunk 1]
```diff
@@ -278,33 +279,35 @@ def test_unique_equal_nan(self, xp, dtype, equal_nan):
...
+    def test_v2api_unique_all(self, xp, dtype, attr):
         a = testing.shaped_random((100, 100), xp, dtype)
         return getattr(xp.unique_all(a), attr)
 
     @testing.with_requires("numpy>=2.0")
     @pytest.mark.parametrize("attr", ["values", "counts"])
     @testing.for_all_dtypes(no_float16=True, no_bool=True, no_complex=True)
     @testing.numpy_cupy_array_equal()
-    def test_unique_counts(self, xp, dtype, attr):
+    def test_v2api_unique_counts(self, xp, dtype, attr):
```
Comment (asi1024) [inline tests/cupy_tests/manipulation_tests/test_add_remove.py:295]: Could you restore the name of the test to `test_unique_counts` and rename the tests for the legacy API to `test_return_unique_counts`?

Comment (EarlMilktea) [reply tests/cupy_tests/manipulation_tests/test_add_remove.py:295]: How about `test_unique_return_counts` ?

I feel this naming is more consistent with other tests/argument names.

Comment (asi1024) [reply tests/cupy_tests/manipulation_tests/test_add_remove.py:295]: Sorry, it's a typo. `test_unique_return_counts` is preferred.

Comment (EarlMilktea) [reply tests/cupy_tests/manipulation_tests/test_add_remove.py]: Fixed in 249bb1f09af708ad2abb9033d9ee79b7712c474e .

**Review by asi1024 — APPROVED** [review]

**Pull request merged**